blitz_api/tests/tests_view_Users.py

```diff
@@ -0,0 +1,610 @@
+import json
+
+from unittest import mock
+
+from rest_framework import status
+from rest_framework.test import APIClient, APITestCase
+
+from django.urls import reverse
+from django.test.utils import override_settings
+from django.contrib.auth import get_user_model
+
+from ..factories import UserFactory, AdminFactory
+from ..models import (ActionToken, Organization, Domain,
+                      AcademicField, AcademicLevel)
+
+User = get_user_model()
+
+
+class UsersTests(APITestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(UsersTests, cls).setUpClass()
+        org = Organization.objects.create(name="random_university")
+        Domain.objects.create(
+            name="mailinator.com",
+            organization_id=org.id
+        )
+        AcademicField.objects.create(name="random_field")
+        AcademicLevel.objects.create(name="random_level")
+
+    def setUp(self):
+        self.client = APIClient()
+
+        self.user = UserFactory()
+        self.user.set_password('Test123!')
+        self.user.save()
+
+        self.admin = AdminFactory()
+        self.admin.set_password('Test123!')
+        self.admin.save()
+
+    def test_create_new_user(self):
+        """
+        Ensure we can create a new user if we have the permission.
+        """
+        data = {
+            'username': 'John',
+            'email': 'John@mailinator.com',
+            'password': 'test123!',
+            'phone': '1234567890',
+            'first_name': 'Chuck',
+            'last_name': 'Norris',
+            'university': {
+                'name': "random_university"
+            },
+            'academic_field': {'name': "random_field"},
+            'academic_level': {'name': "random_level"},
```

Comment:
Can't be more easy to just get an ID here ? 
User list all fields with the API, select one, return it's ID.

Reply:
Finally it's not too hard. I think it's ok  :)

Reply:
We could return their id too if it makes the API easier to use. To make it even easier, I plan on using HyperLinkedSerializer everywhere, so that the API returns the URL for the details of the object.

The real question is how should we handle nested objects? Should we show all the details, only a part with an url and/or id, or url/id only? What is a frontend expecting in general?

Reply:
Front-end expect to have a cohérent content depending on the task we want to do. If some ressources are useless without their nested it become bad to always launch a series of request just to display the ressource.

It depend on the usecase.